src/power_grid_model/validation/utils.py

```diff
@@ -162,7 +168,7 @@ def errors_to_string(
     return msg
 
 
-def nan_type(component: str | ComponentType, field: str, data_type: DatasetType = DatasetType.input):
+def nan_type(component: str | ComponentTypeLike, field: str, data_type: DatasetType = DatasetType.input):
```

Comment:
Shouldn't this just be 
```
def nan_type(component: ComponentTypeLike, field: str, data_type: DatasetType = DatasetType.input):
```

Reply:
yes

Reply:
Resolve when you make the change. I won't so you don't forget.